test/app-tap/module_api.c

```diff
@@ -536,7 +536,9 @@ key_part_padding_offset(void)
 static uint32_t
```

Comment:
Shouldn't we verify something about `box_tuple_compare()` and `box_tuple_compare_with_key()`? AFAIU, their logic depends on the flag.

Reply:
Thank you!

Two nits regarding the new test:

* I would check results for `< 0`, `> 0` instead of equivalence to `-1`, `1`.
* I would call `box_tuple_unref()` at the end.

Reply:
Thanks, I forgot about the `box_tuple_unref`.

Done.